src/v/cloud_storage/topic_manifest.cc

```diff
@@ -118,7 +118,8 @@ struct topic_manifest_handler
                 // Just leave them empty.
                 _properties.segment_size = std::nullopt;
             } else if (_key == "retention_bytes") {
-                _properties.retention_bytes = tristate<size_t>{};
```

Comment:
this seems like a no-op for me, what I'm missing? we even have a test for it

```cpp
BOOST_AUTO_TEST_CASE(disabled_tristate) {
    // disabled
    tristate<int> disabled;
    BOOST_REQUIRE_EQUAL(disabled.is_disabled(), true);
    BOOST_REQUIRE_EQUAL(disabled.has_optional_value(), false);
}
```

Reply:
readability mostly, i needed it [here](https://github.com/redpanda-data/redpanda/pull/14355/commits/519044cda3e96b05486fd70302c531db6b88a0f4#diff-7f0e5af0835b6e2c4f84c11db2287ba29abbf233bf5f473ad6330966555cf856R208-R211) to make it clear what the fix is, and it seems a shame not to use it at least in this file